app/src/main/java/com/example/sharewhatyoucanproject/LoginActivity.kt

```diff
@@ -62,170 +56,155 @@ class LoginActivity : AppCompatActivity() {
         namearray.add("William")
         namearray.add("Henry")
         namearray.add("Benjamin")
-
-
         submitdevice = findViewById(R.id.submitdevice)
-
-        submitdevice.setOnClickListener(View.OnClickListener {
-            deviceid = Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)
-            email = "s" + deviceid + "@gmail.com"
-
-
-
-            pd.show()
-            checkuser()
-
-        })
-
-
+        submitdevice.setOnClickListener(
+            View.OnClickListener {
+                deviceid = Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)
+                email = "s" + deviceid + "@gmail.com"
```

Comment:
I still don't understand why you are making up email address that don't actually exist.  We can't actually email anyone at this address.  Does Firebase require an email address?